src/Resources/views/CRUD/base_array_macro.html.twig

```diff
@@ -8,17 +8,75 @@ For the full copyright and license information, please view the LICENSE
 file that was distributed with this source code.
 
 #}
-{% macro render_array(value, inline) %}
-    {% from _self import render_array %}
-    {% for key, val in value %}
-        {% if val is iterable %}
-            [{{ key }} => {{ render_array(val, inline) }}]
+{# NEXT_MAJOR: move inline parameter under options #}
+{% macro render_array(value, inline, options = []) %}
+    {% spaceless %}
+        {% from _self import render_array %}
+
+        {% if not inline %}
+            <ul>
         {% else %}
-            [{{ key }} => {{ val }}]
+            [
         {% endif %}
 
-        {% if not loop.last and not inline %}
-            <br>
+        {% for key, val in value %}
+            {% if val is not iterable and not inline %}
+                <li>
+            {% endif %}
+
+            {% if options.display in ['both', 'keys'] or options.display is not defined %}
+                {% if options.key_translation_domain is same as(false) or options.key_translation_domain is not defined %}
+                    {{ key }}
+                {% else %}
+                    {% if options.key_translation_domain is same as(true) %}
+                        {% set key_translation_domain = options.default_translation_domain %}
+                    {% elseif options.key_translation_domain is same as(null) %}
+                        {% set key_translation_domain = null %}
+                    {% else %}
+                        {% set key_translation_domain = options.key_translation_domain %}
+                    {% endif %}
+
+                    {{ (options.translation_prefix ~ 'key.' ~ key)|trans({}, key_translation_domain) }}
```

Comment:
I still disagree with the translation_prefix, IMHO I would expect
```suggestion
                    {{ key|trans({}, key_translation_domain) }}
```

When we use the `show_trans` template, there is no prefix added.
If someone use in database
- `thing_to_translate_1`,
- `thing_to_translate_2`,
- ...

He will already have the `thing_to_translate_1`, `thing_to_translate_2`, ... keys in the `messages.yml` so he doesn't want to create `admin_foo.key.thing_to_translate_1`, ... and `admin_bar.key.thing_to_translate_1` etc.

If someone wanted this, I would recommend
`->add(customOptions, array, ...)`
with a getter
```
function getCustomOptions() {
 return array_map($this->options, function(...) {});
}
```